### ISSUE TITLE
Error spam in log file: An exception occurred during UpdateAsync

### ISSUE THREAD

**Issue by romgerman** [FilUnderscore/ImprovedHordes#29]
**Describe the bug**
It started on a blood moon event. My log file has gotten up to 150MB and was growing larger and larger.
```
2024-01-14T23:49:48 87490.750 ERR [Improved Hordes] [WorldHordeTracker] UpdateThread(Action,Func`1): An exception occurred during UpdateAsync: Object reference not set to an instance of an object 
Stacktrace: 
  at ImprovedHordes.Core.World.Horde.WorldHordeTracker+<>c__DisplayClass38_0.<UpdateHorde>b__0 (ImprovedHordes.Core.World.Event.WorldEventReportEvent report) [0x00000] in /var/jenkins_home/workspace/ImprovedHordes_stable/Source/ImprovedHordes/Core/World/Horde/WorldHordeTracker.cs:471 
  at System.Linq.Enumerable+WhereListIterator`1[TSource].MoveNext () [0x00037] in <b222d7464ce94b88b2a6d383411e95be>:0 
  at System.Linq.Enumerable.Any[TSource] (System.Collections.Generic.IEnumerable`1[T] source) [0x00015] in <b222d7464ce94b88b2a6d383411e95be>:0 
  at ImprovedHordes.Core.World.Horde.WorldHordeTracker.UpdateHorde (ImprovedHordes.Core.World.Horde.WorldHorde horde, System.Single dt, System.Collections.Generic.List`1[T] playerGroups, System.Collections.Generic.List`1[T] eventReports) [0x001a0] in /var/jenkins_home/workspace/ImprovedHordes_stable/Source/ImprovedHordes/Core/World/Horde/WorldHordeTracker.cs:474 
  at ImprovedHordes.Core.World.Horde.WorldHordeTracker.UpdateTrackerAsync (System.Collections.Generic.List`1[T] playerHordeGroups, System.Collections.Generic.List`1[T] eventReports, System.Single dt) [0x00016] in /var/jenkins_home/workspace/ImprovedHordes_stable/Source/ImprovedHordes/Core/World/Horde/WorldHordeTracker.cs:490 
  at ImprovedHordes.Core.World.Horde.WorldHordeTracker.UpdateAsync (System.Single dt) [0x0001c] in /var/jenkins_home/workspace/ImprovedHordes_stable/Source/ImprovedHordes/Core/World/Horde/WorldHordeTracker.cs:291 
  at ImprovedHordes.Core.Threading.Threaded.UpdateThread (System.Action resetTime, System.Func`1[TResult] getDeltaTime) [0x00049] in /var/jenkins_home/workspace/ImprovedHordes_stable/Source/ImprovedHordes/Core/Threading/Threaded.cs:83 
```

**To Reproduce**
Not sure

**Expected behavior**
No errors

**Screenshots**
Not needed

**Link to game output log (can be found in 7DaysToDie_Data folder)**
[log snippet.txt](https://github.com/FilUnderscore/ImprovedHordes/files/13933362/log.snippet.txt)

**Additional context**
Idk

**Comment by FilUnderscore** [comment]
Thanks for reporting this bug, it seems to be related to heatmap events not being correctly handled by the mod in some cases. A temporary workaround when encountering this error in the meantime might be to restart the game.

**Comment by FilUnderscore** [comment]
Just to provide an update on this issue, this has been fixed in the latest experimental build for 1.0 experimental: 2.0.1-RC.1. Please let me know if you encounter this issue again.
